pegasus/src/homepage.rb

```diff
@@ -107,6 +107,7 @@ def self.get_heroes
   def self.get_actions(request)
     code_break_takeover = promote_code_break(request)
     code_bytes_takeover = promote_code_bytes(request)
+    hoc2020_ai_takeover = promote_hoc2020_ai(request)
```

Comment:
unless there's a specific reason we do it this way, I think these should all be checking `show_single_hero`, because the way this method is written, if the clauses are in the wrong order, we might get the actions for a previous hero if the flag is still set to true, because everywhere else we use `show_single_hero` as the source of truth.

Maybe a follow up fix

Reply:
I included this change in [this PR](https://github.com/code-dot-org/code-dot-org/pull/38045)